tools/clang/unittests/HLSLTestLib/DxcTestUtils.cpp

```diff
@@ -195,6 +195,77 @@ void AssembleToContainer(dxc::DxcDllSupport &dllSupport, IDxcBlob *pModule,
   CheckOperationSucceeded(pResult, pContainer);
 }
 
+void ReplaceText(llvm::ArrayRef<LPCSTR> pLookFors,
+                llvm::ArrayRef<LPCSTR> pReplacements,
+                bool bRegex, std::string& disassembly) {
+  for (unsigned i = 0; i < pLookFors.size(); ++i) {
+    LPCSTR pLookFor = pLookFors[i];
+    bool bOptional = false;
+    if (pLookFor[0] == '?') {
+      bOptional = true;
+      pLookFor++;
+    }
+    LPCSTR pReplacement = pReplacements[i];
+    if (pLookFor && *pLookFor) {
+      if (bRegex) {
+        llvm::Regex RE(pLookFor);
+        std::string reErrors;
+        if (!RE.isValid(reErrors)) {
+          WEX::Logging::Log::Comment(WEX::Common::String().Format(
+              L"Regex errors:\r\n%.*S\r\nWhile compiling expression '%S'",
+              (unsigned)reErrors.size(), reErrors.data(),
+              pLookFor));
+        }
+        VERIFY_IS_TRUE(RE.isValid(reErrors));
+        std::string replaced = RE.sub(pReplacement, disassembly, &reErrors);
+        if (!bOptional) {
+          if (!reErrors.empty()) {
+            WEX::Logging::Log::Comment(WEX::Common::String().Format(
+                L"Regex errors:\r\n%.*S\r\nWhile searching for '%S' in text:\r\n%.*S",
+                (unsigned)reErrors.size(), reErrors.data(),
+                pLookFor,
+                (unsigned)disassembly.size(), disassembly.data()));
+          }
+          VERIFY_ARE_NOT_EQUAL(disassembly, replaced);
+          VERIFY_IS_TRUE(reErrors.empty());
+        }
+        disassembly = std::move(replaced);
+      } else {
+        bool found = false;
+        size_t pos = 0;
+        size_t lookForLen = strlen(pLookFor);
+        size_t replaceLen = strlen(pReplacement);
+        for (;;) {
+          pos = disassembly.find(pLookFor, pos);
+          if (pos == std::string::npos)
+            break;
+          found = true; // at least once
+          disassembly.replace(pos, lookForLen, pReplacement);
+          pos += replaceLen;
+        }
+        if (!bOptional) {
+          if (!found) {
+            WEX::Logging::Log::Comment(WEX::Common::String().Format(
+                L"String not found: '%S' in text:\r\n%.*S",
+                pLookFor,
+                (unsigned)disassembly.size(), disassembly.data()));
+          }
+          VERIFY_IS_TRUE(found);
+        }
+      }
+    }
+  }
+}
+
+void ReplaceDisassemblyText(llvm::ArrayRef<LPCSTR> pLookFors,
```

Comment:
This overload is odd.  It's only used by one caller.  Can we just move the blob creation into the caller and get rid of this one?  In fact, the caller itself (`CompileAndRewriteAssemblyToText`) only has one caller, so that call could be replaced with the compile, disassemble, ReplaceText, and a CreateBlobFromPinned, and the `CompileAndRewriteAssemblyToText` function can also be eliminated.